pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -328,6 +331,47 @@ func buildNetworkTopologyMetadata(deviceID string, store *metadata.Store) []meta
 	return links
 }
 
+func resolveLocalInterface(deviceID string, interfaceIndexByIDType map[string]map[string]int32, localInterfaceIDType string, localInterfaceID string) (string, string) {
+	if localInterfaceID == "" {
+		return localInterfaceIDType, localInterfaceID
+	}
+
+	var typesToTry []string
+	if localInterfaceIDType == "" {
+		// "smart resolution" by multiple types when localInterfaceIDType is not provided (which is often the case)
+		// CAVEAT: in case the smart resolution returns false positives, the solution is to configure the device to provide a proper localInterfaceIDType
+		typesToTry = []string{"interface_name", "mac_address", "interface_name", "interface_alias", "interface_index"}
+	} else {
+		typesToTry = []string{localInterfaceIDType}
+	}
+	for _, idType := range typesToTry {
+		interfaceIndexByIDValue, ok := interfaceIndexByIDType[idType]
+		if ok {
+			ifIndex, ok := interfaceIndexByIDValue[localInterfaceID]
+			if ok {
+				return metadata.IDTypeNDM, deviceID + ":" + strconv.Itoa(int(ifIndex))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you considered converting to string using `fmt.Sprint` instead of using concatenation and the double conversion int32 -> int -> string ?

Reply:
@pducolin Usually, when the concatenation is simple and doesn't hinder readability I prefer more basic tools like string concat / strconv.Itoa since those requires less CPU cycles than `Sprintf`.

In this case, using either one shouldn't impact much. So, I don't have very strong opinion on either one.